### PR TITLE
Docs: Add more fields to bug report template

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Add a few more fields to the bug report template: one for operating system (since that is sometimes the issue)  and asking for explicit reproduction steps. I also updated the chat link to point directly to the help channel and clarified that we want people to copy-paste the actual ESLint output (since people seem not to do that).

#### Is there anything you'd like reviewers to focus on?

Do these changes make sense?

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ assignees: ''
 * **ESLint Version:**
 * **Node Version:**
 * **npm Version:**
+* **Operating System:**
```
Comment (aladdin-add) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md:34]: currently, running `eslint --env-info` does not print the OS; but it seems good to have.

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 2]
```diff
@@ -61,7 +62,16 @@ assignees: ''
 **What did you expect to happen?**
 
 
-**What actually happened? Please include the actual, raw output from ESLint.**
+**What actually happened? Please copy-paste the actual, raw output from ESLint.**
```
Comment (mdjermanovic) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md:65]: :+1:

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md [hunk 3]
```diff
@@ -61,7 +62,16 @@ assignees: ''
...
 
 
+**Steps to reproduce this issue:**
+
+<!-- Please tell us exactly how to see the issue you're describing -->
+
+1. 
+1. 
+1.
```
Comment (mdjermanovic) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.md:74]: This is sometimes useful, but I think far more often redundant with other fields. Can we instead ask for the exact steps only when we are unable to reproduce the issue?

Comment (nzakas) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.md:74]: I’d rather a bit of redundancy upfront. When I’m looking at bugs, too often people leave out repro steps and not enough people ask. Let’s just get it out of the way early.

Comment (mdjermanovic) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.md:74]: I'm not seeing many issues where additional info is needed, aside from those where people don't fill out the existing sections or completely ignore the template.

What would be repro steps for a bug report related to a rule?

Comment (nzakas) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.md:74]: 1. Type this code into editor “”
2. Configure rule with these options
3. Run ESLint
4. See unexpected rule result

Can you explain your concern over adding this? To me, I’d just like KOR info upfront as I don’t feel like we always get the best bug reports. Especially if we are going to start letting non-team members so triage, I think having this info will really help.

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md (hunk 1 above)
Comment (nzakas) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.md:34]: I’m assuming people know what operating system they are using. :)

File: .github/ISSUE_TEMPLATE/BUG_REPORT.md (hunk 3 above)
Comment (mdjermanovic) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.md:74]: > Can you explain your concern over adding this?

Only that a template with many fields might be discouraging for users who'd like to report a bug.

Comment (nzakas) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.md:74]: I can understand that. My feeling is that this additional ask isn’t enough to dissuade people from filing an issue.

can we try this and see what happens? We can always remove it if it’s a problem, but I think this is important to include to help out the triage team, who may not be as familiar with ESLint internals.

Comment (mdjermanovic) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.md:74]: Agreed!

**Review by mdjermanovic — APPROVED** [review]

**Pull request merged**